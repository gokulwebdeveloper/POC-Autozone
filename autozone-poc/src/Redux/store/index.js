import {createStore, applyMiddleware, compose} from 'redux';
import rootReducer from '../reducers';
import rootSaga from '../middlewares/saga';
import createSagaMiddleware from 'redux-saga';

// create the saga middleware
export const sagaMiddleware = createSagaMiddleware();

//list of middlewares
const middlewares = [sagaMiddleware];

//apply middleware on store
const store = createStore(rootReducer, applyMiddleware(...middlewares));

//run saga generator function
sagaMiddleware.run(rootSaga);

export default store;