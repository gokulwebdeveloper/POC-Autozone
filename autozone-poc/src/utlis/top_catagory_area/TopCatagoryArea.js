import React from 'react';
import PropTypes from 'prop-types';

export default function TopCatagoryArea({topCatagoryAreaText, style}) {
    return (
            <div className="col-12 col-sm-6 col-md-4">
                <div className="single_catagory_area d-flex align-items-center justify-content-center bg-img" style={style}>

                    <div className="catagory-content">
                        <a href="#">{topCatagoryAreaText}</a>
                    </div>
                </div>
            </div>
    )
}

TopCatagoryArea.propTypes = {
    btnText: PropTypes.oneOfType([
        PropTypes.string,
        PropTypes.number])
}

TopCatagoryArea.defaultProps = {
    text: 'topCatagoryAreaText'
}    