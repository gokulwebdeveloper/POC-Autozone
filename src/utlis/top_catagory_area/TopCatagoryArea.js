import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

export default function TopCatagoryArea(props) {
    const bgImgCatImg = {
        'backgroundImage': `url(${props.bgStyleImg})`
      }
    return (
            <div className="col-12 col-sm-6 col-md-4">
                <div className="single_catagory_area d-flex align-items-center justify-content-center bg-img" style={bgImgCatImg}>
                    <div className="catagory-content">
                    <Link  to={{
                        pathname:'/shops',
                        filter: {
                        filterType: props.topCatagoryAreaText,
                        filter: 'All'
                        }
                      }}>{props.topCatagoryAreaText}</Link >
                    </div>
                </div>
            </div>
    )
}

TopCatagoryArea.propTypes = {
    topCatagoryAreaText: PropTypes.oneOfType([
        PropTypes.string,
        PropTypes.number])
}

TopCatagoryArea.defaultProps = {
    topCatagoryAreaText: 'topCatagoryAreaText'
}