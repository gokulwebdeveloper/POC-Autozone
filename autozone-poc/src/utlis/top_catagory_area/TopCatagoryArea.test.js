import React from 'react';
import '@testing-library/jest-dom/extend-expect';
import { configure, mount, shallow } from 'enzyme';
import renderer from "react-test-renderer";
import Adapter from 'enzyme-adapter-react-16';
import '../../matchMedia.mock'
import TopCatagoryArea from './TopCatagoryArea';
import store from '../../Redux/store';
import { Provider } from 'react-redux';

configure({ adapter: new Adapter() });

describe('TopCatagoryArea page testing', () => {
  //mount component for testing
  beforeEach(() => {
  });

  let wrapper = mount(<Provider store={store}><TopCatagoryArea /></Provider>);

  //test if TopCatagoryArea component exists
  it("TopCatagoryArea Exists", () => {
    expect(wrapper.exists()).toBeTruthy();
  });

  // Render the a tag
  test('renders the a tag', () => {
    const link = wrapper.find("NavLink");
    expect(link).toHaveLength(0);
  });

  const wrappersnap = shallow(<Singleblog />);
  it("SnapShotÂ Match", () => {
    expect(wrappersnap).toMatchSnapshot();
  });

  //SnapShot Match test
  it("SnapShot Match", () => {
    const topCatagoryAreaComponent = renderer.create(<TopCatagoryArea />).toJSON();
    expect(topCatagoryAreaComponent).toMatchSnapshot();
  });
})