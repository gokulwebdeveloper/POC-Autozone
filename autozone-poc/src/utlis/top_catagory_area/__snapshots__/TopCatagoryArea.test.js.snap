// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopCatagoryArea page testing SnapShot Match 1`] = `
<div
  className="col-12 col-sm-6 col-md-4"
>
  <div
    className="single_catagory_area d-flex align-items-center justify-content-center bg-img"
  >
    <div
      className="catagory-content"
    >
      <a
        href="#"
      >
        topCatagoryAreaText
      </a>
    </div>
  </div>
</div>
`;
